guillotina_batch/api.py

```diff
@@ -106,15 +106,15 @@ class Batch(Service):
                 pass
         return request
 
-    async def handle(self, message):
+    async def handle(self, message, request_headers=None):
         payload = message.get('payload') or {}
         if not isinstance(payload, str):
             payload = ujson.dumps(payload)
         request = await self.clone_request(
             message['method'],
             message['endpoint'],
             payload,
-            message.get('headers') or {})
+            message.get('headers') or request_headers or {})
```

Comment:
I think instead of doing an `or` here, we should do something like:

``` 
headers = dict(self.request.headers)
headers.update(message.get('headers') or {})
```

so the message headers would just be providing new ones or overriding headers.